Classes/Package/CustomPages.php

```diff
@@ -164,49 +162,44 @@ public function optionsPage() {
 		}
 	}
 
-	public function footerJs() {
+	public function pageState( $post_states, $post ) {
 
 		if ( function_exists( 'get_field' ) ) {
-			$screen = get_current_screen();
-
-			if ( 'page' == $screen->id ) {
-				$page_pt = false;
-				foreach ( $this->getPosttypeItems() as $pt => $name ) {
-					$field = get_field( "page_for_$pt", 'options' );
-					if ( $field && $field == $_GET['post'] ) {
-						$page_pt = $name;
+			$items = array_merge( $this->getPosttypeItems(), $this->getSpecialItems() );
+
+			foreach ( $items as $key => $name ) {
+				$page_id = get_field( "page_for_$key", 'options' );
+
+				if ( intval( $page_id ) == $post->ID ) {
+					if ( get_post_type_object( $key ) ) {
+						$post_states[] = sprintf( __( 'Archive Page for "%s"', 'sha' ), $name );
+					} else {
+						$post_states[] = sprintf( __( 'Page for "%s"', 'sha' ), $name );
```

Comment:
Begriffe bitte sprachlich einheitlichen.

```suggestion
						$post_states[] = sprintf( __( 'Seite für «%s»', 'sha' ), $name );
```